packages/Telephony/Common/CommandDialog{T}.cs

```diff
@@ -54,30 +62,51 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
 
         public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default)
         {
-            // check activity
             var activity = dc.Context.Activity;
 
-            // If command result, handle it
+            // We are expecting a command result with the same name as our current CommandName.
             if (activity.Type == ActivityTypes.CommandResult
-                && activity.Name == this.Name.GetValue(dc.State))
+                && activity.Name == this.CommandName.GetValue(dc.State))
             {
                 // TODO: correlate command id before handling it.
 
                 var commandResult = TelephonyExtensions.GetCommandResultValue(activity);
 
-                if (commandResult.Error != null)
+                if (commandResult?.Error != null)
                 {
                     throw new ErrorResponseException($"{commandResult.Error.Code}: {commandResult.Error.Message}");
                 }
 
                 return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
             }
 
-            // for now, end turn and keep waiting
-            // TODO: Carlos add interruption model
+            // This activity was not the command result we were expecting. Mark as waiting and end the turn.
             return new DialogTurnResult(DialogTurnStatus.Waiting);
         }
 
+        /// <inheritdoc/>
+        protected override async Task<bool> OnPreBubbleEventAsync(DialogContext dc, DialogEvent e, CancellationToken cancellationToken)
+        {
+            if (e.Name == DialogEvents.ActivityReceived && dc.Context.Activity.Type == ActivityTypes.Message)
+            {
+                // Ask parent to perform recognition
```

Comment:
I noticed that the recognition block is before the interruption check. Will recognize utterance lead to activities being processed?

Reply:
We are just emitting the event but if we return false in this method, then interruptions will not block this flow no matter the recognition.